js/documents.js

```diff
@@ -353,54 +336,49 @@ var documentsMain = {
 			$('#revisionsContainer ul').append(html);
 		},
 
-		fetchAndFillRevisions: function(documentPath) {
-			// fill #rev-history with file versions
-			// FIXME: this is no longer supported in OC10
-			$.get(OC.generateUrl('apps/files_versions/ajax/getVersions.php?source={documentPath}&start={start}',
-			                     { documentPath: documentPath, start: documentsMain.UI.revisionsStart }),
-				  function(result) {
-					  for(var key in result.data.versions) {
-						  documentsMain.UI.addRevision(documentsMain.fileId,
-						                               result.data.versions[key].version,
-						                               result.data.versions[key].humanReadableTimestamp,
-						                               documentPath);
-					  }
-
-					  // owncloud only gives 5 version at max in one go
-					  documentsMain.UI.revisionsStart += 5;
-
-					  if (result.data.endReached) {
-						  // Remove 'More versions' button
-						  $('#show-more-versions').addClass('hidden');
-					  }
-				  });
-		},
-
 		showRevHistory: function(documentPath) {
 			$(document.body).prepend(documentsMain.UI.viewContainer);
 
 			var revHistoryContainerTemplate = Handlebars.compile(documentsMain.UI.revHistoryContainerTemplate);
 			var revHistoryContainer = revHistoryContainerTemplate({
 				filename: documentsMain.fileName,
-				moreVersionsLabel: t('richdocuments', 'More versions...'),
 				closeButtonUrl: OC.imagePath('core', 'actions/close')
 			});
 			$('#revViewerContainer').prepend(revHistoryContainer);
 
-			documentsMain.UI.revisionsStart = 0;
+			// show loading screen
+			$('#revPanelContainer').hide();
+			documentsMain.overlay.documentOverlay('show');
 
 			// append current document first
-			documentsMain.UI.addRevision(documentsMain.fileId, 0, t('richdocuments', 'Just now'), documentPath);
+			documentsMain.UI.addRevision(documentsMain.fileId, 0, t('richdocuments', 'Just now'));
+			
+			// load revisions
+			$.getJSON(
+				OC.generateUrl('apps/richdocuments/ajax/documents/revisions/{fileId}', {
+					fileId: documentsMain.fileId
+				})
+			).done(function (result) {
+				// add revisions
+				for(var key in result.revisions) {
+					documentsMain.UI.addRevision(
+						documentsMain.fileId,
+						result.revisions[key].version,
+						result.revisions[key].humanReadableTimestamp
+					);
+				}
+			}).fail(function(result){
+				console.error(result);
+				documentsMain.UI.notify(t('richdocuments', 'Cannot load document revisions'));
+			}).always(function(){
+				// show revisions container after loading
+				documentsMain.overlay.documentOverlay('hide');
+				$('#revPanelContainer').show();
 
-			// add "Show more versions" button
-			$('#show-more-versions').click(function(e) {
-				e.preventDefault();
-				documentsMain.UI.fetchAndFillRevisions(documentPath);
+				// fake click on first revision (i.e current revision)
```

Comment:
what if you have multiple revisions and you're on the third one? Not sure when the `showRevHistory` is called, but going back to the first revision if you just want to show the revisions doesn't look a good behavior.

Reply:
This is how collabora works, a bit weird. if this fake click is not done, you would have blank screen and then would need to select version.